hrjobcontract/tests/phpunit/HRJobContractTestTrait.php

```diff
@@ -44,6 +44,55 @@ protected function createJobContract($contactID, $startDate = null, $endDate = n
   }
 
   /**
+   * Creates a new Job Contract for the given contact
+   *
+   * This is a little bit different than createJobContract
+   * its needed to test getCurrentContract
+   * TODO: change createJobContract behavior and remove this
+   * @param array $params should contain contact_id, start_date, end_date, position, title
+   * @throws \CiviCRM_API3_Exception
+   */
+  protected function createContract($params) {
+    $HRJobContract = civicrm_api3('HRJobContract', 'create', array(
+      'contact_id' => $params['contact_id'],
+    ));
+    $HRJobContractRevision = civicrm_api3('HRJobContractRevision', 'create', array(
+      'jobcontract_id' => $HRJobContract['id'],
+      'effective_date' => $params['start_date'],
+      'effective_end_date' => $params['end_date'],
+    ));
+    civicrm_api3('HRJobContractRevision', 'create', array(
+      'id' => $HRJobContractRevision['id'],
+      'details_revision_id' => $HRJobContractRevision['id'],
+    ));
+    civicrm_api3('HRJobDetails', 'create', array(
+      'jobcontract_id' => $HRJobContract['id'],
+      'jobcontract_revision_id' => $HRJobContractRevision['id'],
+      'position' => $params['position'],
+      'title' => $params['title'],
+      'period_start_date' => $params['start_date'],
+      'period_end_date' => $params['end_date'],
+    ));
```

Comment:
Since we're inside the hrjobcontract extension, why use the API? I think it's a little weird to use the API in this situation and it's certainly slower than calling the BAOs directly.